zkvm/src/predicate.rs

```diff
@@ -68,66 +90,26 @@ impl Predicate {
         encoding::write_point(&self.to_point(), prog);
     }
 
-    /// Verifies whether the current predicate is a disjunction of n others.
-    /// Returns a `PointOp` instance that can be verified in a batch with other operations.
-    pub fn prove_disjunction(&self, preds: &[Predicate]) -> PointOp {
-        let f = Self::commit_disjunction(preds.iter().map(|p| p.to_point()));
-
-        // P = X[0] + f*B
-        PointOp {
-            primary: Some(f),
-            secondary: None,
-            arbitrary: vec![
-                (Scalar::one(), preds[0].to_point()),
-                (-Scalar::one(), self.to_point()),
-            ],
-        }
-    }
-
-    /// Verifies whether the current predicate is a commitment to a program `prog`.
-    /// Returns a `PointOp` instance that can be verified in a batch with other operations.
-    pub fn prove_program_predicate(&self, prog: &[u8], blinding: &[u8]) -> PointOp {
-        let h = Self::commit_program(prog, blinding);
-        // P == h*B2   ->   0 == -P + h*B2
-        PointOp {
-            primary: None,
-            secondary: Some(h),
-            arbitrary: vec![(-Scalar::one(), self.to_point())],
-        }
-    }
-
-    /// Creates a program with a random blinding factor.
-    pub fn blinded_program<T: Into<Program>>(x: T) -> Self {
-        let blinding: [u8; 16] = rand::random();
-        Predicate::Program(x.into(), blinding.to_vec())
-    }
-
-    /// Creates a program with an empty blinding factor.
-    pub fn unblinded_program<T: Into<Program>>(x: T) -> Self {
-        Predicate::Program(x.into(), Vec::new())
-    }
-
     /// Downcasts the predicate to a verification key
     pub fn to_key(self) -> Result<VerificationKey, VMError> {
         match self {
             Predicate::Key(k) => Ok(k),
+            Predicate::Tree(t) => {
+                let ovk = VerificationKey::from_compressed(t.precomputed_point);
```

Comment:
you can use `.ok_or(VMError::InvalidPoint)` which does the same job as your match statement.